### ISSUE TITLE
Vitest restore mock sets all hoisted mocks to undefined when following documentation

### ISSUE THREAD

**Issue by rstainsby** [nuxt/test-utils#864]
### Environment

- Operating System: Linux
- Node Version:     v18.20.3
- Nuxt Version:     3.11.2
- CLI Version:      3.11.1
- Nitro Version:    2.9.6
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      devtools, modules
- Runtime Modules:  @nuxt/test-utils/module@3.13.1
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-htfm4n?file=README.md

### Describe the bug

Nuxt's testing documentation recommends restoring mocks that you have hoisted using vi.hoisted() https://nuxt.com/docs/getting-started/testing#built-in-mocks.

The approach the documentation recommends for this is:

```js
const { useStorageMock } = vi.hoisted(() => {
    return { useStorageMock: vi.fn().mockImplementation(() => {
      return { value: 'mocked storage'}
    })
  }
})

mockNuxtImport('useStorage', () => {
  return useStorageMock
})

// Then, inside a test
useStorageMock.mockImplementation(() => {
  return { value: 'something else' }
})
```

but when using vitests mock restore, this results in useStorageMock being immediately restored to undefined. So, unless you mock useStorageMock in every test, it will likely cause broken tests. This doesn't feel like the implied behavior - In the example above I expected restoring the mock to reset 'value' to 'mocked storage'. 

This happens because mockRestore will restore a mocked function (`vi.fn()`) to undefined if that mocked function doesn't have an initial value (`vi.fn(initalval)`). Docs on that are here: https://vitest.dev/api/mock.html#mockrestore.

The solution for this then seems to be to define a initial value for the mocked function. So I propose that the docs are changed to:

```js
const { useStorageMock } = vi.hoisted(() => {
    return { 
        useStorageMock: vi.fn(() => {
          return { value: 'mocked storage' }
        })
     }
})

mockNuxtImport('useStorage', () => {
  return useStorageMock
})

// Then, inside a test
useStorageMock.mockImplementation(() => {
  return { value: 'something else' }
})

```

I've included a stackblitz with tests that showcase this. More detail is included in the README of the stackblitz.

### Additional context

TLDR: If you follow the docs recommendation of calling mockRestore before each test, you will end up with undefined mocks for every mocked Nuxt import that requires different implementations between tests.
